apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/CcwRifLoadJob.java

```diff
@@ -370,32 +379,28 @@ private boolean dataSetIsAvailable(DataSetManifest manifest) {
         String.format(
             "%s/%s/", manifest.getManifestKeyIncomingLocation(), manifest.getTimestampText());
 
-    ListObjectsV2Request s3BucketListRequest = new ListObjectsV2Request();
-    s3BucketListRequest.setBucketName(options.getS3BucketName());
-    s3BucketListRequest.setPrefix(dataSetKeyPrefix);
-    if (options.getS3ListMaxKeys().isPresent())
-      s3BucketListRequest.setMaxKeys(options.getS3ListMaxKeys().get());
+    ListObjectsV2Request.Builder s3BucketListRequestBuilder =
+        ListObjectsV2Request.builder().bucket(options.getS3BucketName()).prefix(dataSetKeyPrefix);
+    if (options.getS3ListMaxKeys().isPresent()) {
+      s3BucketListRequestBuilder.maxKeys(options.getS3ListMaxKeys().get());
+    }
 
     Set<String> dataSetObjectNames = new HashSet<>();
-    ListObjectsV2Result s3ObjectListing;
-    do {
-      s3ObjectListing = s3TaskManager.getS3Client().listObjectsV2(s3BucketListRequest);
+    ListObjectsV2Iterable s3ObjectListingPaginator =
+        s3TaskManager.getS3Client().listObjectsV2Paginator(s3BucketListRequestBuilder.build());
 
-      /*
-       * Pull the object names from the keys that were returned, by
-       * stripping the timestamp prefix and slash from each of them.
-       */
-      Set<String> namesForObjectsInPage =
-          s3ObjectListing.getObjectSummaries().stream()
-              .map(s -> s.getKey())
-              .peek(s -> LOGGER.debug("Found file: '{}', part of data set: '{}'.", s, manifest))
-              .map(k -> k.substring(dataSetKeyPrefix.length()))
-              .collect(Collectors.toSet());
-      dataSetObjectNames.addAll(namesForObjectsInPage);
-
-      // On to the next page! (If any.)
-      s3BucketListRequest.setContinuationToken(s3ObjectListing.getNextContinuationToken());
-    } while (s3ObjectListing.isTruncated());
+    /*
+     * Pull the object names from the keys that were returned, by
+     * stripping the timestamp prefix and slash from each of them.
+     */
+
+    Set<String> namesForObjectsInPage =
+        s3ObjectListingPaginator.stream()
+            .flatMap(s -> s.contents().stream())
+            .peek(s -> LOGGER.debug("Found file: '{}', part of data set: '{}'.", s, manifest))
+            .map(k -> k.key().substring(dataSetKeyPrefix.length()))
```

Comment:
This flatmap is publishing s3 objects now rather than their keys.  That's bad since it could result in the object itself being logged.  I suggest changing it to this:

```
        s3ObjectListingPaginator.stream()
            .flatMap(s -> s.contents().stream())
            .map(o -> o.key())
            .peek(k -> LOGGER.debug("Found file: '{}', part of data set: '{}'.", k, manifest))
            .map(k -> k.substring(dataSetKeyPrefix.length()))
            .collect(Collectors.toSet());
```